src/transformStateWithClones.js

```diff
@@ -7,7 +7,42 @@
  * @return {Object[]}
  */
 function transformStateWithClones(state, actions) {
-  // write code here
+  const stateHistory = []; // will be our result
+  let currentState = { ...state }; /* create copy of the
+  initial object so later we can push more copies of this object
+   to the stateHistory array at each iteration cycle without modifying
+   initial state object */
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties': {
+        currentState = { ...currentState, ...action.extraData };
+        /* if action array has an object with 'type: addProperties' property,
+        we add properties from extraData object and create another copy of
+        current state */
+        break;
+      }
+
+      case 'removeProperties': {
+        currentState = { ...currentState };
+        action.keysToRemove.forEach((key) => delete currentState[key]);
+        /* additional iteration through keysToRemove object to remove keys
+        from currentState */
+        break;
+      }
+
+      case 'clear': {
+        currentState = {};
+        break;
+      }
+    }
```

Comment:
[CODE STYLE]: switch/case should always have a default case for error handling. It's important to handle unexpected action types to ensure that the function behaves predictably even when encountering unknown actions.